client/src/client.rs

```diff
@@ -482,26 +482,54 @@ impl Client {
         }
     }
 
-    // TODO: introduce timeout
     fn listen_for_tx_confirmation(
         &self,
+        init_sender: tokio::sync::oneshot::Sender<bool>,
```

Comment:
I'm a bit ambiguous on this wrapping a `bool`. From a type standpoint, the older variant of sending unit made more sense, as we don't distinguish true and false scenarios here.

Reply:
@ilchu, maybe i understand you wrong, but we handle this scenarios differently.

```rust
let _send_result = init_sender.send(event_iterator_result.is_ok());
```

In case error happened during initialization we send false to channel and don't submit transaction.

Reply:
Right, my bad, then 100% fine